lib/chef/resource.rb

```diff
@@ -1316,7 +1316,11 @@ class << self
     # life as well.
     @@sorted_descendants = nil
     def self.sorted_descendants
-      @@sorted_descendants ||= descendants.sort_by { |x| x.to_s }
+      # so it turns out Class.to_s just blindly returns the name of the class,
+      # which can be a Symbol in addition to being a String.  as a result, we
+      # may get a Symbol back from Class.to_s, so we call #to_s on that again.
+      # (buggy on at least ruby-2.4.4)
+      @@sorted_descendants ||= descendants.sort_by { |x| x.to_s.to_s }
```

Comment:
Isn't it though?  I tore out so much hair this morning I look like I could be a presidental candidate...